src/thunderbots/software/geom/util.cpp

```diff
@@ -752,6 +752,59 @@ std::optional<Point> lineIntersection(const Vector &a, const Vector &b, const Ve
     return std::make_optional(intersection);
 }
 
+std::optional<Point> finiteLengthLineIntersection(const Vector &a, const Vector &b,
+                                                  const Vector &c, const Vector &d)
+{
+    std::optional<Point> intersection = lineIntersection(a, b, c, d);
+
+    if (intersection != std::nullopt)
+    {
+        // We know the point returned by lineIntersection() exists along the trajectory of
+        // one of the lines, now we must check that it is within the specified line points
+        // If point C exists on line A-B, DIST(AC) + DIST(CB) = DIST(AB)
+        // Calculated for lines a-b and c-d
+        if ((fabs(dist(a, *intersection)) + fabs(dist(*intersection, b)) ==
+             fabs(dist(a, b))) &&
+            fabs(dist(c, *intersection)) + fabs(dist(*intersection, d)) ==
+                fabs(dist(c, d)))
+        {
+            return intersection;
+        }
+    }
+
+    return std::nullopt;
+}
+
+std::optional<Point> velocity_line_intersection(const Vector element_velocity,
+                                                const Vector element_position,
+                                                const Point line_start,
+                                                const Point line_end,
+                                                const Rectangle boundry_area)
+{
+    // If the element is not within the boundries, return null
+    if (!boundry_area.containsPoint(element_position))
+    {
+        return std::nullopt;
+    }
+
+    // Calculate the closest edge of the boundry based on element speed and position
+    const double seconds_to_x_field_edge = std::max(
```

Comment:
These variables should have "field" in them. The rectangle may not be referring to the field and this function has no concept of the field.

Reply:
Good catch, I removed all references to time and field.